### PR TITLE
GoSal linux v4 

### PR DESCRIPTION
**Pull request by gavinelder**
# Add support for linux with v4

- Updated readme removed references to Dep
- **Breaking** Removed unused `management` from Config & code as these are unused
- Parity with windows GoSal
 
## Testing

- Spun up SalV4 from latest docker confirmed checkin and relevant metadata was correct. 

image

## Drivebys 

- Linting
- Cleanup some err's

### CODE REVIEW

File: xpreports/build_linux.go [hunk 1]
```diff
@@ -1,12 +1,100 @@
...
+
+	consoleUser, _ := linux.GetLoggedInUsers()
+
+	v, _ := mem.VirtualMemory()
+
+	computerSystem, _ := linux.GetLinuxComputerSystem()
+
+	// Convert memory from kb to correct size
+	convertedMemory := float64(v.Total)
+	unitCount := 0
```
Comment (groob) [inline xpreports/build_linux.go]: Prefer `var foo Int` or `var foo string` for zero value.

File: xpreports/build_linux.go [hunk 2]
```diff
@@ -1,12 +1,100 @@
...
+	strMemory := ""
+
+	for convertedMemory >= 1024 {
+		convertedMemory = convertedMemory / 1024
+		unitCount++
+	}
+
+	switch unitCount {
```
Comment (groob) [inline xpreports/build_linux.go]: This style of formatting is more appropriate for a server to do (if possible)

**Comment by bdemetris** [comment]
sorry I haven't gotten to this.  Thanks @groob for taking a look (and also reminding me!)

We don't use this for linux or have any immediate plans to do so.  The project being used to report the info seems legit, and it had windows support?  pretty neat!

@gavinelder if you can just add some #TODO hashes on the error handling stuff that would be helpful.  If we start using this on linux (crostini is a strong possibility) it will at least remind me to circle back on that stuff.

**Comment by gavinelder** [comment]
Have added some super basic error handling for this now, messages are not the most verbose but it's there.

**Pull request merged**